app/session.server.ts

```diff
@@ -43,9 +45,47 @@ export async function getUsername(request: Request): Promise<User['username'] |
   return username;
 }
 
+export async function getEffectiveUsername(
+  request: Request
+): Promise<User['username'] | undefined> {
+  const cookie = request.headers.get('Cookie');
+  const effectiveUsername = await effectiveUsernameCookie.parse(cookie);
+  return effectiveUsername as User['username'];
+}
+
+export async function getEffectiveUser(request: Request) {
+  const username = await getUsername(request);
+  // get the effective username in the cookie at the time
+  const effectiveUsername = await getEffectiveUsername(request);
+  if (!username) {
+    return null;
+  }
+
+  // if there is no effective username let the admin be himself
+  if (!effectiveUsername || effectiveUsername === '') {
```

Comment:
It makes me nervous that `effectiveUsername` could be `''`.  When could this happen and why?  Can we avoid that?

Reply:
This was what I remembered from our discussion, admins would have an empty effectiveUsername until they select one

Reply:
Can "empty" be `undefined` vs `''`?